packages/comms/utils/index.ts

```diff
@@ -0,0 +1,253 @@
+                }
+            }
+        }
+    }
+
+    if (knownTypes.indexOf(opName) < 0) {
+        knownTypes.push(opName);
+        parsedTypes[opName] = typeDefn;
+    }
+    return typeDefn;
+}
+
+wsdlToTs(args.url)
+    .then(clientObjs => {
+        const [wsdl, descr] = clientObjs;
+        const bindings = wsdl.definitions.bindings;
+        const wsdlNS = wsdl.definitions.$targetNamespace;
+        let namespace = "";
+        for (const ns in descr) {
+            namespace = changeCase(ns, Case.PascalCase);
+            const service = descr[ns];
+            printDbg("namespace: ", namespace, "\n");
+            for (const op in service) {
+                printDbg("binding: ", changeCase(op, Case.PascalCase), "\n");
+                const binding = service[op];
+                for (const svc in binding) {
+                    const operation = binding[svc];
+                    const types = wsdl.definitions.schemas[wsdlNS].types;
+                    const request = operation["input"];
+                    const reqName = bindings[op].methods[svc].input.$name;
+                    const response = operation["output"];
+                    const respName = bindings[op].methods[svc].output.$name;
+
+                    parseTypeDefinition(request, reqName, types);
+                    parseTypeDefinition(response, respName, types);
+                }
+            }
+        }
+
+        const relativePath = path.relative(path.join(cwd, outDir), path.join(cwd, "./src")).replace(/\\/g, "/");
+        lines.push(`import { IConnection, IOptions } from "${relativePath}/connection";`);
+        lines.push(`import { Service } from "${relativePath}/espConnection";`);
+        lines.push("\n\n");
+
+        knownPrimitives.forEach(primitive => {
+            lines.push(`type ${primitive} = ${primitiveMap[primitive]};`);
+        });
+        lines.push("\n\n");
+
+        for (const name in parsedEnums) {
+            lines.push(`export enum ${name} {\n\n`);
+            lines.push(parsedEnums[name].join(",\n"));
+            lines.push("\n\n}");
+            lines.push("\n\n");
+        }
+
+        lines.push(`export namespace ${namespace} {\n`);
+
+        for (const type in parsedTypes) {
+            lines.push(`export interface ${type} {\n`);
+            const typeString = JSON.stringify(parsedTypes[type], null, 4).replace(/"/g, "");
+            lines.push(typeString.substring(1, typeString.length - 1) + "\n");
+            lines.push("}\n");
+        }
+
+        lines.push("}");
+
+        lines.push("\n\n");
+
+        lines.push(`export class ${namespace.replace("Ws", "")}Service extends Service {\n`);
+
+        const methods = [];
+
+        for (const service in bindings) {
+            const binding = bindings[service];
+            for (const method in binding.methods) {
+                const soapAction = binding.methods[method].soapAction;
+                const url = `http://localhost:8010/${soapAction}`;
```

Comment:
`http://localhost:8010` should be extracted from the URL passed in as an argument?

Reply:
Added a comment and changed to "example.org" instead